File: backend/src/main/java/com/finalproject/pms/service/AdminService.java
package com.finalproject.pms.service;

import com.finalproject.pms.dto.admin.AdminAnalyticsResponse;
import com.finalproject.pms.dto.auth.UserResponse;
import com.finalproject.pms.dto.parking.ParkingSummaryResponse;

import com.finalproject.pms.dto.booking.BookingResponse;
import com.finalproject.pms.dto.review.ReviewResponse;

import java.util.List;

public interface AdminService {
    List<UserResponse> getUsers();
    AdminAnalyticsResponse getAnalytics();
    List<BookingResponse> getAllBookings();
    List<ReviewResponse> getAllReviews();
    ParkingSummaryResponse approveParking(Long parkingId);
    void deactivateUser(Long id);
    void activateUser(Long id);
}

File: frontend/src/services/adminService.js
import { api } from './api'

export function getUsers() {
  return api.get('/api/admin/users').then((response) => response.data)
}

export function getAnalytics() {
  return api.get('/api/admin/analytics').then((response) => response.data)
}

export function getAllBookings() {
  return api.get('/api/admin/bookings').then((response) => response.data)
}

export function getAllReviews() {
  return api.get('/api/admin/reviews').then((response) => response.data)
}

export function approveParking(parkingId) {
  return api.put(`/api/admin/parking/${parkingId}/approve`).then((response) => response.data)
}

export function deleteReview(reviewId) {
  return api.delete(`/api/admin/reviews/${reviewId}`)
}

export function deactivateUser(userId) {
  return api.put(`/api/admin/users/${userId}/deactivate`).then((response) => response.data)
}

export function activateUser(userId) {
  return api.put(`/api/admin/users/${userId}/activate`).then((response) => response.data)
}
